CenterForOpenScience/ember-osf-web
# [ENG-4187] Test file detail view section - Updated

-   Ticket: https://openscience.atlassian.net/browse/ENG-4187
-   Branch: feature/file-metadata-mirage-tests

## Purpose

The purpose of these changes is to add tests to ember-osf-web for the File Metadata project.

## Summary of Changes

Tests were added to the registration-file-detail-test.ts file to verify the functionality of metadata label and div elements, the edit metadata form and its save and cancel buttons, and the mobile default view for the application.

## Screenshot(s)

<img width="1200" alt="Pasted Graphic 29" src="https://user-images.githubusercontent.com/82047646/219084844-13411076-da2c-4369-a116-498be8ea6ccf.png">

## Side Effects

There should be no unintended side effects.

## QA Notes

-This PR references the following PRs that were revised: https://github.com/CenterForOpenScience/ember-osf-web/pull/1756 && https://github.com/CenterForOpenScience/ember-osf-web/pull/1793. Any changes reflected here should also include changes referenced in them.
-Are all added components and functionalities properly tested?
-Are there any suggested tests not already covered?
-Does the default view change for tablet view? From tablet to mobile, mobile to desktop and vice versa?
-Are all elements properly rendered on the screen? Does the edit metadata button remain hidden when permissions change?
